src/main/java/org/czekalski/petapiintegration/resourceserver/controller/AnimalController.java

```diff
@@ -0,0 +1,34 @@
+package org.czekalski.petapiintegration.resourceserver.controller;
+
+import org.czekalski.petapiintegration.config.AnimalRepresentationModelAssembler;
+import org.czekalski.petapiintegration.resourceserver.apiresource.v1.dto.AnimalResourcesListDto;
+import org.czekalski.petapiintegration.resourceserver.service.AnimalService;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+public class AnimalController {
+    private final AnimalService animalService;
+    private static final AnimalRepresentationModelAssembler animalRepresentationModelAssembler = new AnimalRepresentationModelAssembler();
```

Comment:
To powinno być przez konstruktor. Podczas testowania kodu oczywiście, że nie musisz tego mockować, tylko wrzucić gotową instancję :) W ten sposób jesteś zgodny z Dependency Inversion. Klasa `AnimalController` będzie niezależna od implementacji `AnimalRepresentationModelAssembler`. Teraz, gdy chciałbyś zmienić coś z `AnimalRepresentationModelAssembler`, np. dodać polę do konstruktora to musiałbyś przerabiać klasę `AnimalController` również.

Reply:
Poprawione: f45d5f1b33b01d46ad17c7f91f352bae0aca3402